### PR TITLE
Ddpb 3732

### PR DESCRIPTION
**Pull request by jamesrwarren**
## Purpose
Adding basic scaling functionality to Frontend and API so that we can handle sudden increases in load.

Fixes DDPB-3732

## Approach
Steal LPAs implementation because it is running in live currently and I know that Andrew did some good testing on it.

## Learning
NA

## Checklist
- [x] I have performed a self-review of my own code
- [ ] I have updated documentation (Confluence/ADR/tech debt doc) where relevant
- [ ] I have added tests to prove my work
- [ ] The product team have approved these changes

### Frontend
- [ ] I have run an in-browser accessibility test (e.g. WAVE, Lighthouse)
- [ ] There are no deprecated CSS classes noted in the profiler
- [ ] Translations are used and the profiler doesn't identify any missing
- [ ] Any links or buttons added are screen reader friendly and contextually complete

### CODE REVIEW

File: environment/ecs_autoscaling/main.tf [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  metric_name               = "RunningTaskCount"
+  namespace                 = "ECS/ContainerInsights"
+  period                    = "30"
+  statistic                 = "Average"
+  threshold                 = var.ecs_task_autoscaling_maximum
+  alarm_description         = "This metric monitors ecs running task count for the ${var.environment} ${var.aws_ecs_service_name} service"
+  insufficient_data_actions = []
+  dimensions = {
+    ServiceName = var.aws_ecs_service_name
+    ClusterName = "${var.environment}-online-lpa"
```
Comment (sixdaysandy) [inline environment/ecs_autoscaling/main.tf:59]: This isn't the cluster that you're looking for 😄

File: environment/ecs_autoscaling/main.tf [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    }
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "max_scaling_reached" {
+  alarm_name                = "${var.environment}-${var.aws_ecs_service_name}-max-scaling-reached"
+  comparison_operator       = "GreaterThanOrEqualToThreshold"
+  evaluation_periods        = "2"
+  metric_name               = "RunningTaskCount"
+  namespace                 = "ECS/ContainerInsights"
```
Comment (sixdaysandy) [inline environment/ecs_autoscaling/main.tf:51]: This won't work in prod just yet

**Review by sixdaysandy — CHANGES_REQUESTED** [review]

File: environment/ecs_autoscaling/main.tf (hunk 1 above)
Comment (jamesrwarren) [reply environment/ecs_autoscaling/main.tf]: OMG.. yes!

File: environment/ecs_autoscaling/main.tf (hunk 2 above)
Comment (jamesrwarren) [reply environment/ecs_autoscaling/main.tf:51]: Yes true. Won't merge till it's active

**Review by sixdaysandy — DISMISSED** [review]

**Review by sixdaysandy — DISMISSED** [review]

**Review by acsauk — APPROVED** [review]
:shipit:

**Review by sixdaysandy — APPROVED** [review]

**Pull request merged**